### PR TITLE
Datasource read error propagation

### PR DESCRIPTION
**Pull request by jsantos17**
[ch9584]

### CODE REVIEW

File: impl/src/main/scala/quasar/impl/push/DefaultResultPush.scala [hunk 1]
```diff
@@ -78,12 +78,12 @@ class DefaultResultPush[
       query = tableRef.query
       columns = tableRef.columns
 
-      evaluated <- EitherT.rightT(format match {
```
Comment (jsantos17) [inline impl/src/main/scala/quasar/impl/push/DefaultResultPush.scala]: `QueryEvaluator`'s `evaluate` function returns a `F[Stream[F, Byte]]`. Before this change, we sequenced the outer `F` right here, so any errors produced within that `F` would not be caught. Now the errors are handled asynchrnously by `JobManager`.

Comment (alissapajer) [reply impl/src/main/scala/quasar/impl/push/DefaultResultPush.scala]: Are there any tests we can add that would show that an errors are now caught?

Comment (jsantos17) [reply impl/src/main/scala/quasar/impl/push/DefaultResultPush.scala]: Oh that's a good observation. Will add!

Comment (jsantos17) [reply impl/src/main/scala/quasar/impl/push/DefaultResultPush.scala]: Done. So this should be ready for another look.

**Pull request merged**